discovery_engine_core/core/src/engine.rs

```diff
@@ -360,20 +374,25 @@ impl Engine<xayn_ai::ranker::Ranker> {
                 .map_err(|err| Error::Ranker(err.into()))?
                 .build()
                 .map_err(|err| Error::Ranker(err.into()))?;
-            Engine::from_state(&state.engine, config.into(), ranker, stacks_ops).await
+            Self::from_state(&state.engine, config.into(), ranker, stack_ops).await
         } else {
             let ranker = builder.build().map_err(|err| Error::Ranker(err.into()))?;
-            Engine::new(config.into(), ranker, stacks_ops).await
+            Self::new(config.into(), ranker, stack_ops).await
         }
     }
 }
 
+/// A shared discovery engine with a lock.
+#[allow(clippy::module_name_repetitions)]
+#[derive(AsRef, From)]
+pub struct SharedEngine(Mutex<XaynAiEngine>);
```

Comment:
Is still true that this must stay here for some limitation we found in cbindgen?

Reply:
i'm not entirely sure what the exact issue was, i'll check again if i can move it to the bindgen crate.

Reply:
i moved the `SharedEngine` into the `bindings` crate. to avoid the `unused_qualifications` lint, i placed it into its own submodule instead of the crate root, because we still have to call it with full path within the `async-bindgen` macro. i also renamed the `core` dependency to its original name `xayn-discovery-engine-core` as this otherwise causes name conflicts for example in other proc macros which expect the standard library's `core` module behind this name.